server/v2/cometbft/server.go

```diff
@@ -66,6 +66,11 @@ func (s *CometBFTServer[T]) Init(appI serverv2.AppI[T], v *viper.Viper, logger l
 		AppTomlConfig:    appTomlConfig,
 	}
 
+	appGenesis, err := genutiltypes.AppGenesisFromFile(s.config.ConfigTomlConfig.GenesisFile())
```

Comment:
It's actually this logic we should re-implement here: https://github.com/cosmos/cosmos-sdk/blob/6b7e9f8935e7ec2af61a3b17c3daab24ab0cfde0/server/util.go#L505

Reply:
ah i see ty